soda_machine.rb

```diff
@@ -7,12 +9,25 @@ def initialize(args = {})
   end
 
   def current_inventory_count
+    sodas.count
   end
 
   def find_soda(soda_brand)
+    sodas.each do |soda|
+      return soda if soda.brand == soda_brand
+    end
+    return nil
   end
 
   def sell(soda_brand)
```

Comment:
if the soda isn't there, don't you need to return ``` nil ``` ? If you do, check out this approach: 

```
  def sell(soda_brand)
    return nil if find_soda(soda_brand) == nil
    soda_to_sell = find_soda(soda_brand)
    @cash += soda_to_sell.price
    @sodas.delete(soda_to_sell)
  end
```